### ISSUE TITLE
[Form] Error submiting datetime as a string

### ISSUE THREAD

**Issue by emilien-puget** [symfony/symfony#12114]
With symfony 2.5.5.

When submitting a date like '2015-01-01 00:00', DateTimeToLocalizedStringTransformer crash, but array('2015-01-01 00:00') does not and is intercepted by the validator shortly after.

The date '2015-01-01 00:00:00' works as well as array('2015-01-01 00:00:00').

I fixed this issue by disabling the reverseTransform method of DateTimeToLocalizedStringTransformer
ps : obviously, the 'fix' broke others things

edit : I've found a cleaner fix

I use :

``` php
private function isValidDateTimeString($dateString, $formatString)
    {
        $date = \DateTime::createFromFormat($formatString, $dateString);

        return $date && \DateTime::getLastErrors()["warning_count"] == 0 && \DateTime::getLastErrors(
        )["error_count"] == 0;
    }
```

In the reverseTransform of DateTimeToLocalizedStringTransformer, before the parse (https://github.com/symfony/symfony/blob/3ecf68535cd3b09f881fe7e403a49439675ebef2/src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php#L126) and 

``` php
return '';
```

**Comment by javiereguiluz** [comment]
Closing this ancient bug report because there's no more activity or feedback. Please reopen if you still experience this in a modern and currently supported Symfony version. Thanks!

**Comment by robmeek** [comment]
I’m running PHP 7.4 and Symfony 5.3.9.
I just encountered this ancient bug with a DateTimeType:

```php
$builder->add('date', DateTimeType::class, [
    'widget' => 'single_text',
    'format' => 'yyyy-MM-dd HH:mm',
    'html5' => false,
    'by_reference' => true,
]);
```

Persisting using this form worked fine on my local machine.
But on my production server I got a validation error:

```
This value is not valid
```

Digging deeper in the web profiler I found the same TransformationFailedException described above.

@origaminal’s hints set me on the right path.
**Simply installing the php-intl extension** in production resolved this issue for me – perhaps that will help someone else struggling with this.
